File: Backend/app/api/roadmap.py
from fastapi import APIRouter

from app.schemas.roadmap import LearningRoadmapResponse
from app.services.roadmap_service import RoadmapService

router = APIRouter(
    prefix="/roadmap",
    tags=["Roadmap"],
)


@router.post(
    "/generate",
    response_model=LearningRoadmapResponse,
)
def generate_learning_roadmap(
    skill_report: str,
):
    """
    Generate a personalized learning roadmap
    from the candidate's skill report.
    """

    service = RoadmapService()

    return service.generate_roadmap(
        skill_report
    )

File: frontend/src/api/roadmap.ts
import { apiClient } from './client'
import type { Roadmap } from '@/types'

// POST /roadmap/generate?skill_report=...  — generate a learning roadmap from a skill report text
// The backend expects skill_report as a query parameter
export const generateRoadmap = async (skillReportSummary: string): Promise<Roadmap> => {
  const { data } = await apiClient.post<Roadmap>(
    '/roadmap/generate',
    null,
    { params: { skill_report: skillReportSummary } }
  )
  return data
}
